core/domain/exp_services_test.py

```diff
@@ -881,13 +881,102 @@ def test_export_by_versions(self):
 def _get_change_list(state_name, property_name, new_value):
     """Generates a change list for a single state change."""
     return [{
-        'cmd': 'edit_state_property',
+        'cmd': exp_domain.CMD_EDIT_STATE_PROPERTY,
         'state_name': state_name,
         'property_name': property_name,
         'new_value': new_value
     }]
 
 
+class UpdateGadgetTests(ExplorationServicesUnitTests):
+    """Test updating a gadget."""
+
+    def setUp(self):
+        super(UpdateGadgetTests, self).setUp()
+        exploration = self.save_new_valid_exploration(
+            self.EXP_ID, self.owner_id)
+
+        self.init_state_name = exploration.init_state_name
+
+        self.param_changes = [{
+            'customization_args': {
+                'list_of_values': ['1', '2'], 'parse_with_jinja': False
+            },
+            'name': 'myParam',
+            'generator_id': 'RandomSelector'
+        }]
+        # List of answer groups to add into an interaction.
+        self.interaction_answer_groups = [{
+            'rule_specs': [{
+                'rule_type': 'Equals',
+                'inputs': {'x': 0},
+            }],
+            'outcome': {
+                'dest': self.init_state_name,
+                'feedback': ['Try again'],
+                'param_changes': []
+            },
+        }]
+        # Default outcome specification for an interaction.
+        self.interaction_default_outcome = {
+            'dest': self.init_state_name,
+            'feedback': ['Incorrect', '<b>Wrong answer</b>'],
+            'param_changes': []
+        }
+
+
+    def test_gadget_commands(self):
```

Comment:
Hi @kefeh, I'm afraid that, without further investigation, I don't know the answer to your question.

However, debugging is an important part of programming. I suggest that you look at the stack trace, read through the code, understand what it is doing, and figure out what's going on behind the scenes. You can also log specific variables (or raise them in exceptions) while running the test in order to get a window into what's happening.